### PR TITLE
rgw: metadata sync info should be shown at master zone of slave zoneg…

### PR DESCRIPTION
**Pull request by Jing-Scott**
…roup

When executing 'radosgw-admin sync status', the metadata sync info should be shown on the srceen at master zone of slave zonegroup.

Signed-off-by: Jing Wenjun <jingwenjun@cmss.chinamobile.com>

### CODE REVIEW

**Comment by Jing-Scott** [comment]
@cbodley Please help to have a look. Thanks! ^_^
I have encontered the problem when I created a user in the master zone of master zonegroup, the master zone of slave zonegroup didn't sync the created user actually for other reasons(eg. network). But executing `radosgw-admin sync status`, the metadata sync info shown "no sync (zone is master)". So the metadata sync info is needed for the master zone of secondary zonegroup.

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -1975,7 +1975,7 @@ static void sync_status(Formatter *formatter)
 
   list<string> md_status;
 
-  if (zone.id == zonegroup.master_zone) {
+  if (zonegroup.is_master && zone.id == zonegroup.master_zone) {
```
Comment (cbodley) [inline src/rgw/rgw_admin.cc]: looks right. there's a `store->is_meta_master()` function that does the same thing - could you use that instead?

**Comment by cbodley** [comment]
I created a tracker issue so we can get this fix backported to jewel. could you please add this line to your commit message?
```
Fixes: http://tracker.ceph.com/issues/18091
```

**Comment by Jing-Scott** [comment]
@cbodley Thanks for your suggestion^_^. I have modified it and added the tracker issue line in my commit message.

File: src/rgw/rgw_admin.cc [hunk 2]
```diff
@@ -1975,7 +1975,7 @@ static void sync_status(Formatter *formatter)
 
   list<string> md_status;
 
-  if (zone.id == zonegroup.master_zone) {
+  if (store->is_meta_master() && zone.id == zonegroup.master_zone) {
```
Comment (cbodley) [inline src/rgw/rgw_admin.cc]: sorry, i meant this could just be:
```c++
if (store->is_meta_master()) {
```

**Comment by cbodley** [comment]
thanks! would you mind squashing the two into a single commit? (this makes it easier for the backport team)

**Comment by Jing-Scott** [comment]
@cbodley Okay. I have squashed the two. Thanks!

**Pull request merged**